cluster/service.go

```diff
@@ -30,17 +28,13 @@ type Cluster struct {
 	Token string
 }
 
-func cleanURL(url string) string {
-	if !strings.HasSuffix(url, "/") {
-		return url + "/"
-	}
-	return url
-}
+type GetCluster func(ctx context.Context, target string) (*Cluster, error)
 
 // Service the interface for the cluster service
 type Service interface {
-	GetClusters(context.Context) ([]Cluster, error)
-	Stats() Stats
+	GetCluster(ctx context.Context, target string) (*Cluster, error)
+	GetClusters(ctx context.Context) []Cluster
```

Comment:
for API consistency, you can have GetCluser() return `Cluster` rather `*Cluster`. (nice to have comment)

Reply:
changed in 84fe6dc